### ISSUE TITLE
Bug in reading dielectronic recombination data (.drparams files)

### ISSUE THREAD

**Issue by wtbarnes** [chianti-atomic/ChiantiPy#180]
The drparams file reader has a bug in it where in the case of those rates calculated using the Badnell method (i.e. Eq. 3 of [this paper](http://cdsads.u-strasbg.fr/abs/2003A%26A...412..587Z)), the last entry in the list of fit parameters, for both the E and C fit parameters, is not read. 

For example, for Fe 8, the `fe_8.drparams` file looks like,
```
    1
   26    8  9.4990e+01  3.0310e+02  7.8780e+02  2.0800e+03  7.2090e+03  3.1590e+04  1.0200e+05  2.8030e+05  7.8400e+05
   26    8  3.1130e-07  8.7830e-07  3.9560e-06  5.5390e-05  9.4220e-05  9.3430e-04  5.2960e-03  1.7800e-02  2.2690e-01
```
and the dictionary read by ChiantiPy looks like,
```python
{'cparams': array([  3.11300000e-07,   8.78300000e-07,   3.95600000e-06,
          5.53900000e-05,   9.42200000e-05,   9.34300000e-04,
          5.29600000e-03,   1.78000000e-02]),
 'drtype': 1,
 'eparams': array([  9.49900000e+01,   3.03100000e+02,   7.87800000e+02,
          2.08000000e+03,   7.20900000e+03,   3.15900000e+04,
          1.02000000e+05,   2.80300000e+05])}
```
Note the last entry is missing in the latter case.

This problem can go unnoticed in those cases when the last few fit parameters are 0, but in the case where they are finite, it make a significant difference in the dielectronic recombination rate (and thus the total rate), e.g. for Fe 8,

![image](https://user-images.githubusercontent.com/8676141/32299241-e393fa54-bf22-11e7-92cb-33e411b35722.png)

This also propagates through to the ionization equilibrium populations and makes a significant difference for Fe 7,8,9,

![image](https://user-images.githubusercontent.com/8676141/32299320-31ad0712-bf23-11e7-83ec-4365f4bd5f60.png)

**Comment by kdere** [comment]
I have not checked it but it looks like it definitely needs correcting.  I am a bit busy today so I you want to take your try, go ahead

**Comment by wtbarnes** [comment]
This bug also seems to come up in the [IDL code for calculating the recombination rate](http://www.chiantidatabase.org/idl/ionrec/recomb_rate.pro) though there it seems the number of parameters in the Badnell fit files is hardcoded at 8 (even though there are 9 parameters). Do you know why this is? Is the last parameter purposefully ignored for some reason?

![image](https://user-images.githubusercontent.com/8676141/32339916-2c74882a-bfc7-11e7-8fbe-d2c1b38390a9.png)

**Comment by kdere** [comment]
considering that I wrote that piece of code, I do not know why.  Probably a typical mistake of mine

**Comment by kdere** [comment]
It seems that you have a fix for this.  If you can do a PR, please do so.  I suspect that the limited format is a copy of the IDL code.

**Comment by wtbarnes** [comment]
Sure I can open a PR. Is there a formal channel for reporting bugs in the IDL code? This seems like a pretty significant bug and should probably be fixed in the latest version.

**Comment by wtbarnes** [comment]
See #181. Here's the new comparison. As you can see, there's still an issue with the IDL results.

![image](https://user-images.githubusercontent.com/8676141/32595273-e34fb736-c4f4-11e7-92a3-33d86d492e25.png)

**Comment by kdere** [comment]
First, let's fix the ChiantiPy code and then fix the IDL code.
The only way to fix the IDL code is by one of the members.  It is under SVN..

**Comment by wtbarnes** [comment]
This is fixed by #181. 

Ok. From what I can tell the IDL code is also essentially a 1 character fix.
